toolkit/tools/imagegen/installutils/installutils.go

```diff
@@ -440,6 +441,26 @@ func PopulateInstallRoot(installChroot *safechroot.Chroot, packagesToInstall []s
 		return
 	}
 
+	if isRootFS && (len(config.Users) > 0 || len(config.Groups) > 0) {
```

Comment:
Could we refactor this to only call addUser etc. once, it feels very odd to have two flows here (for rootfs and non-rootfs). I'd assume that non-rootfs images can show the same type of issues with pre-defining users.